nixos/modules/programs/nix-ld.nix

```diff
@@ -5,62 +5,146 @@
   ...
 }:
 let
-  cfg = config.programs.nix-ld;
+  inherit (lib)
+    literalExpression
+    mergeAttrsList
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    mkRenamedOptionModule
+    types
+    ;
 
-  nix-ld-libraries = pkgs.buildEnv {
-    name = "ld-library-path";
-    pathsToLink = [ "/lib" ];
-    paths = map lib.getLib cfg.libraries;
-    # TODO make glibc here configurable?
-    postBuild = ''
-      ln -s ${pkgs.stdenv.cc.bintools.dynamicLinker} $out/share/nix-ld/lib/ld.so
-    '';
-    extraPrefix = "/share/nix-ld";
-    ignoreCollisions = true;
-  };
+  cfg = config.programs.nix-ld.systems;
+
+  share-path = system: "share/nix-ld-${system}";
+
+  nix-ld-libraries =
+    system: cfg:
+    cfg.pkgs.buildEnv {
+      name = "ld-library-path";
+      pathsToLink = [ "/lib" ];
+      paths = map lib.getLib cfg.libraries;
+      # TODO make glibc here configurable?
+      postBuild = ''
+        ln -s ${cfg.pkgs.stdenv.cc.bintools.dynamicLinker} $out/${share-path system}/lib/ld.so
+      '';
+      extraPrefix = "/${share-path system}";
+      ignoreCollisions = true;
+    };
 in
 {
   meta.maintainers = [ lib.maintainers.mic92 ];
-  options.programs.nix-ld = {
-    enable = lib.mkEnableOption ''nix-ld, Documentation: <https://github.com/nix-community/nix-ld>'';
-    package = lib.mkPackageOption pkgs "nix-ld" { };
-    libraries = lib.mkOption {
-      type = lib.types.listOf lib.types.package;
-      description = "Libraries that automatically become available to all programs. The default set includes common libraries.";
-      default = [ ];
-      defaultText = lib.literalExpression "baseLibraries derived from systemd and nix dependencies.";
-    };
-  };
 
-  config = lib.mkIf config.programs.nix-ld.enable {
-    environment.ldso = "${cfg.package}/libexec/nix-ld";
+  imports =
+    let
+      system = pkgs.system or ''''${pkgs.system}'';
+    in
+    [
+      (mkRenamedOptionModule
```

Comment:
I would prefer to make `"programs" "nix-ld" "enable"` an alias instead of deprecating it, because it makes it easier to enable nix-ld across architectures.

Reply:
Done, apparently `mkAliasOptionModuleMD` is itself no longer needed and just an alias for `mkAliasOptionModule`

Reply:
Thank you, this does resolve the issue about the deprecation warning, and I do agree it makes it more portable. Now the remaining question is about the defaults